### PR TITLE
GAC Test Harness

### PR DESCRIPTION
**Pull request by hunterbarber**
## Fixes/Resolves:
GAC in #1302

## Summary/Motivation:
Consolidating GAC testing using new `UnitTestHarness`

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by hunterbarber** [comment]
I think the best thing to do is try to get this merged as the GAC unit test harness PR, and then have subsequent PR's for the costing test as we decide what we're doing with them.

**Review by MarcusHolly — APPROVED** [review]
LGTM - I agree that costing tests can be expanded upon later in the `watertap.costing.unit_models.tests` directory. My only comment is that I'm unsure if the test for reporting is necessary in the unit model test - although it can't hurt.

**Review by savannahsakhai — APPROVED** [review]
LGTM. I agree that its best to leave the costing for another PR.

**Review by bknueven — CHANGES_REQUESTED** [review]
We should merge #1317 first as it has needed API changes for the `UnitTestHarness`.

File: watertap/unit_models/tests/test_gac.py [hunk 1]
```diff
@@ -13,646 +13,307 @@
...
+    iscale.calculate_scaling_factors(m)
+
+    return m
+
+
+class TestGACHand(UnitTestHarness):
+    def configure(self):
+        m = build_hand()
+
+        self.unit_model_block = m.fs.unit
```
Comment (bknueven) [inline watertap/unit_models/tests/test_gac.py]: This is now optional, but it doesn't hurt.

File: watertap/unit_models/tests/test_gac.py [hunk 2]
```diff
@@ -13,646 +13,307 @@
...
+        self.unit_solutions[m.fs.unit.equil_conc] = 0.0005178
+        self.unit_solutions[m.fs.unit.dg] = 19780
+        self.unit_solutions[m.fs.unit.N_Bi] = 6.113
+        self.unit_solutions[m.fs.unit.min_N_St] = 35.68
+        self.unit_solutions[m.fs.unit.throughput] = 0.9882
+        self.unit_solutions[m.fs.unit.min_residence_time] = 468.4
+        self.unit_solutions[m.fs.unit.residence_time] = 134.7
+        self.unit_solutions[m.fs.unit.min_operational_time] = 9153000
+        self.unit_solutions[m.fs.unit.operational_time] = 2554000
+        self.unit_solutions[m.fs.unit.bed_volumes_treated] = 8514
```
Comment (bknueven) [inline watertap/unit_models/tests/test_gac.py:113]:
```suggestion
        self.unit_solutions[m.fs.unit.bed_volumes_treated] = 8514
        
        return m
```

File: watertap/unit_models/tests/test_gac.py [hunk 3]
```diff
@@ -13,646 +13,307 @@
...
+        self.unit_solutions[m.fs.unit.operational_time] = 13690000
+        self.unit_solutions[m.fs.unit.bed_volumes_treated] = 22810
+        self.unit_solutions[m.fs.unit.velocity_int] = 0.003157
+        self.unit_solutions[m.fs.unit.bed_length] = 0.8333
+        self.unit_solutions[m.fs.unit.bed_area] = 10.68
+        self.unit_solutions[m.fs.unit.bed_volume] = 8.900
+        self.unit_solutions[m.fs.unit.bed_diameter] = 3.688
+        self.unit_solutions[m.fs.unit.bed_mass_gac] = 4004
+        self.unit_solutions[m.fs.unit.conc_ratio_avg] = 0.2287
+        self.unit_solutions[m.fs.unit.ele_operational_time[1]] = 6462000
```
Comment (bknueven) [inline watertap/unit_models/tests/test_gac.py:207]:
```suggestion
        self.unit_solutions[m.fs.unit.ele_operational_time[1]] = 6462000
        
        return m
```

File: watertap/unit_models/tests/test_gac.py [hunk 4]
```diff
@@ -13,646 +13,307 @@
...
+        self.unit_model_block = m.fs.unit
+
+        # arguments for UnitTestHarness
+        self.default_zero = zero
+        self.default_relative_tolerance = relative_tolerance
+
+        self.unit_solutions[m.fs.unit.N_Re] = 2.473
+        self.unit_solutions[m.fs.unit.N_Sc] = 2001
+        self.unit_solutions[m.fs.unit.kf] = 2.600e-5
+        self.unit_solutions[m.fs.unit.ds] = 1.245e-14
```
Comment (bknueven) [inline watertap/unit_models/tests/test_gac.py:306]:
```suggestion
        self.unit_solutions[m.fs.unit.ds] = 1.245e-14
        
        return m
```

**Review by bknueven — APPROVED** [review]

**Pull request merged**